optuna/samplers/_grid.py

```diff
@@ -20,7 +20,7 @@
 
 
 GridValueType = Union[str, float, int, bool, None]
-SortableParamValueSequenceType = Union[List[str], List[float], List[int], List[bool]]
+SortableParamValueSequenceType = Sequence[GridValueType]
```

Comment:
Since `List` is not the same as the type of `choices` in `CateoricalDistriibution`, `Sequence` is supposed to be more accurate.